circuit_maintenance_parser/parsers/aws.py

```diff
@@ -23,10 +23,26 @@ def parse_subject(self, subject):
 
         Example: AWS Direct Connect Planned Maintenance Notification [AWS Account: 00000001]
         """
-        data = {}
-        search = re.search(r"\[AWS Account ?I?D?: ([0-9]+)\]", subject)
-        if search:
-            data["account"] = search.group(1)
+        data = {"account": ""}
```

Comment:
If we can't find the "account", we should skip adding the key with an empty value. It's better to fail clearly than have a result like the one in `tests/unit/data/aws/aws3_subject_parser_result.json`

Reply:
I guess that's the rub in this case. Some of the emails have the account number in the subject line, while others don't, (it's in the body).  If I just skip adding an empty key, there is no other data thats parsed on subject line, and it returns an empty data structure `[{}]` this breaks the upstream parser tho, as it considers it to have failed parsing vs. an expected empty response.  Maybe I'm not understanding it clearly tho' - what would be a good practice to do here when you are EXPECTING there to be no data under certain conditions?

Reply:
the `Providers` (e.g., aws), contain different `Processors`, which are a combination of one or more `Parsers`.
The way the circuit maintenance parser works is to try to extract part of the `Maintenance` data from each `Parser` and then finally combine the data to create a valid `Maintenance`.
This means that is some `Processor` has a `Parser` that is not capturing the proper data it's ok, another one could do in the same process.

Reply:
Alright cool, I think after incorporating the feedback you gave me, I think we probably are ready to give this a look over again. Let me know what you think! 👍